config/source/300-rabbitmqsource.yaml

```diff
@@ -1,51 +1,290 @@
+                      description: Type refers to the CloudEvent type attribute.
+                      type: string
+                  type: object
+                type: array
+              conditions:
+                description: Conditions the latest available observations of a resource's
+                  current state.
+                items:
+                  description: 'Condition defines a readiness condition for a Knative
+                    resource. See: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties'
+                  properties:
+                    lastTransitionTime:
+                      description: LastTransitionTime is the last time the condition
+                        transitioned from one status to another. We use VolatileTime
+                        in place of metav1.Time to exclude this from creating equality.Semantic
+                        differences (all other things held constant).
+                      type: string
+                    message:
+                      description: A human readable message indicating details about
+                        the transition.
+                      type: string
+                    reason:
+                      description: The reason for the condition's last transition.
+                      type: string
+                    severity:
+                      description: Severity with which to treat failures of this type
+                        of condition. When this is not specified, it defaults to Error.
+                      type: string
+                    status:
+                      description: Status of the condition, one of True, False, Unknown.
+                      type: string
+                    type:
+                      description: Type of condition.
+                      type: string
+                  required:
+                  - status
+                  - type
+                  type: object
+                type: array
+              observedGeneration:
+                description: ObservedGeneration is the 'Generation' of the Service
+                  that was last processed by the controller.
+                format: int64
+                type: integer
+              sinkUri:
+                description: SinkURI is the current active sink URI that has been
+                  configured for the Source.
+                type: string
+            type: object
+        type: object
     served: true
     storage: true
     subresources:
       status: {}
-    schema:
-      openAPIV3Schema:
-        type: object
-        # Workaround, existing schema is incomplete and fails validation.
-        x-kubernetes-preserve-unknown-fields: true
-  names:
-    categories:
-      - all
-      - knative
-      - eventing
-      - sources
-      - importers
-    kind: RabbitmqSource
-    plural: rabbitmqsources
-  scope: Namespaced
+status:
```

Comment:
@ChunyiLyu do we need this status?

Reply:
Created an issue for it already: https://github.com/knative-sandbox/eventing-rabbitmq/issues/593
This PR will focus on migrating to controller-gen to keep it concise and ideally merged quickly.